hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/states/Node2PipelineMap.java

```diff
@@ -71,6 +71,10 @@ public synchronized void addPipeline(Pipeline pipeline) {
       UUID dnId = details.getUuid();
       dn2ObjectMap.computeIfAbsent(dnId, k -> ConcurrentHashMap.newKeySet())
           .add(pipeline.getId());
+      dn2ObjectMap.computeIfPresent(dnId, (k, v) -> {
```

Comment:
Line 73 should be able to add(pipeline.getId() if the HashMap exist. Line 74-77 seems redundant.

Reply:
From Java doc:https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function- , it looks like computeIfAbsent is only adding absent member. Line 74 is trying to merge a pipelineId into an existed candidate. This happens when one datanode is assigned to multiple pipelines, which is a classic scenario for multiraft.